autorest/preparer_test.go

```diff
@@ -70,7 +70,7 @@ func ExampleCreatePreparer() {
 	} else {
 		fmt.Println(r.URL)
 	}
-	// Output: https://microsoft.com/a/b/c
```

Comment:
Removed url.Opaque. It was giving invalid url string. URL string does not fetch host from url.Host when opaque is non-empty and probably expects everything in opaque. Not sure.
```
http:/a/b/c
```

I have updated WithPathParameters() to fix this. Instead of using WithPath() to concatenate path  and then WithPathParameters() for substituting the parameters, we will only use WithPathParameters() when we have path with parameters and it will build the path and parse to url.

```Go
//For URL: http://microsoft.com/a/{b}/c with b = xyz.

Prepare(..
           WithBaseURI("http://microsoft.com")  // --> url : http://microsoft.com
           WithPathParameters("/a/{b}/c",{"b":"xyz"}))   // --> url: http:/microsoft.com/a/xyz/c

//For simple URL with no parameters in path:   http://microsoft.com/m/n

Prepare(..
           WithBaseURI("http://microsoft.com")  // --> url : http://microsoft.com
           WithPath("/m/n")     // --> //url: http:/microsoft.com/m/n
```